node/src/components/storage.rs

```diff
@@ -712,6 +751,37 @@ impl Storage {
             .transpose()
     }
 
+    pub fn get_highest_block_header_for_db_check(&self) -> Option<Block> {
+        let mut txn = self.env.begin_ro_txn().ok()?;
+        if let Some(height) = self.block_height_index.keys().last() {
+            self.get_block_by_height(&mut txn, *height).ok()?
+        } else {
+            None
+        }
+    }
+
+    pub fn get_state_root_hashes_for_trie_check(&self) -> Option<Vec<Blake2bHash>> {
+        let mut blake_hashes: Vec<Blake2bHash> = Vec::new();
+        let header_txn = match self.env.begin_ro_txn() {
```

Comment:
This is just a `txn`, there's nothing specific to `header` about it.

Reply:
Renamed to `txn`